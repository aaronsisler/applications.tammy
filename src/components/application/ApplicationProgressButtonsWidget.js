import React from 'react';
import PropTypes from 'prop-types';

export default class ApplicationProgressButtonsWidget extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="application_progress_buttons_widget">
                <div className="application_progress_buttons_wrapper">
                    {this.props.currentStep > 0 &&
                        <button id='previousStep' className="button" onClick={this.props.handleDecrementCurrentStep}>Previous Step</button>
                    }
                    {this.props.currentStep < (this.props.maxSteps - 2) &&
                        <button id='nextStep' className="button" onClick={this.props.handleIncrementCurrentStep}>Save and Continue</button>
                    }
                    {this.props.currentStep == (this.props.maxSteps - 2) &&
                        <button id='submitApplication' className="button" onClick={this.props.handleIncrementCurrentStep}>Submit Application</button>
                    }
                </div>
            </div>
        );
    }
}

ApplicationProgressButtonsWidget.propTypes = {
    currentStep: PropTypes.number.isRequired,
    maxSteps: PropTypes.number.isRequired,
    handleDecrementCurrentStep: PropTypes.func.isRequired,
    handleIncrementCurrentStep: PropTypes.func.isRequired,
};
