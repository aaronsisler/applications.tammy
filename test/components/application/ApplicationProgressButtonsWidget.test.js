import React from 'react';
import { shallow } from 'enzyme';
import ApplicationProgressButtonsWidget from 'Application/ApplicationProgressButtonsWidget';

describe('ApplicationProgressButtonsWidget', () => {
    let wrapper;
    const maxSteps = 4;
    const handleDecrementCurrentStep = jest.fn();
    const handleIncrementCurrentStep = jest.fn();

    const buildWrapper = (currentStep = 0) => {
        wrapper = shallow(
            <ApplicationProgressButtonsWidget
                currentStep={currentStep}
                maxSteps={maxSteps}
                handleDecrementCurrentStep={handleDecrementCurrentStep}
                handleIncrementCurrentStep={handleIncrementCurrentStep}
            />);
    };

    describe('when at first step', () => {
        beforeEach(() => {
            buildWrapper();
        });

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('clicking button calls handleIncrementCurrentStep from props', () => {
            wrapper.find('#nextStep').simulate('click');
            expect(handleIncrementCurrentStep).toHaveBeenCalled();
        });
    });

    describe('when at a middle step', () => {
        beforeEach(() => {
            buildWrapper(1);
        });

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('clicking previous step button calls handleDecrementCurrentStep from props', () => {
            wrapper.find('#previousStep').simulate('click');
            expect(handleDecrementCurrentStep).toHaveBeenCalled();
        });

        it('clicking next step button calls handleIncrementCurrentStep from props', () => {
            wrapper.find('#nextStep').simulate('click');
            expect(handleIncrementCurrentStep).toHaveBeenCalled();
        });
    });

    describe('when at final step', () => {
        beforeEach(() => {
            buildWrapper(maxSteps - 2);
        });

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('clicking previous step button calls handleDecrementCurrentStep from props', () => {
            wrapper.find('#previousStep').simulate('click');
            expect(handleDecrementCurrentStep).toHaveBeenCalled();
        });

        it('clicking submit application button calls handleIncrementCurrentStep from props', () => {
            wrapper.find('#submitApplication').simulate('click');
            expect(handleIncrementCurrentStep).toHaveBeenCalled();
        });
    });
})
