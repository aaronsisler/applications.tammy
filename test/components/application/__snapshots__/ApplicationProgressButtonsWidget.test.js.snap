// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationProgressButtonsWidget when at a middle step renders correctly 1`] = `
<div
  className="application_progress_buttons_widget"
>
  <div
    className="application_progress_buttons_wrapper"
  >
    <button
      className="button"
      id="previousStep"
      onClick={[MockFunction]}
    >
      Previous Step
    </button>
    <button
      className="button"
      id="nextStep"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      Save and Continue
    </button>
  </div>
</div>
`;

exports[`ApplicationProgressButtonsWidget when at final step renders correctly 1`] = `
<div
  className="application_progress_buttons_widget"
>
  <div
    className="application_progress_buttons_wrapper"
  >
    <button
      className="button"
      id="previousStep"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      Previous Step
    </button>
    <button
      className="button"
      id="submitApplication"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      Submit Application
    </button>
  </div>
</div>
`;

exports[`ApplicationProgressButtonsWidget when at first step renders correctly 1`] = `
<div
  className="application_progress_buttons_widget"
>
  <div
    className="application_progress_buttons_wrapper"
  >
    <button
      className="button"
      id="nextStep"
      onClick={[MockFunction]}
    >
      Save and Continue
    </button>
  </div>
</div>
`;
